src/main/java/bio/terra/cli/context/GlobalContext.java

```diff
@@ -170,43 +165,82 @@ public void updateDockerImageId(String dockerImageId) {
   //       - sub-directory for persisting pet SA keys: pet-keys/[terra user id]
   //           - pet SA key filename: [workspace id]
 
-  /** Getter for the global context directory. */
-  public static Path resolveGlobalContextDir() {
-    // TODO: allow overriding the global context directory path (e.g. env var?)
-    return DEFAULT_GLOBAL_CONTEXT_DIR;
+  /**
+   * Get the global context directory path relative to the user home directory.
+   *
+   * @return absolute path to global context directory, relative to the user home directory
+   */
+  @JsonIgnore
+  public static Path getGlobalContextDir() {
+    // TODO: allow overriding the global context directory path (e.g. env var != user home
+    // directory)
+    return getGlobalContextDir(Paths.get(System.getProperty("user.home")));
   }
 
   /**
-   * Getter for the sub-directory of the global context directory that holds the pet SA key files
-   * for all users.
+   * Get the global context directory path relative to the given directory.
+   *
+   * @param relativeToDir the top-level directory (e.g. $HOME on either the host or container)
+   * @return absolute path to the global context directory, relative to the given directory
    */
-  private static Path resolvePetSaKeyDir() {
-    return resolveGlobalContextDir().resolve(PET_KEYS_DIRNAME);
+  @JsonIgnore
+  public static Path getGlobalContextDir(Path relativeToDir) {
+    return relativeToDir.resolve(GLOBAL_CONTEXT_DIRNAME).toAbsolutePath();
   }
 
   /**
-   * Getter for the sub-directory of the global context directory that holds the pet SA key files
-   * for the given user.
+   * Get a handle to the global context file.
```

Comment:
what's the difference between a file and a file handle?

Reply:
There's an interesting challenge her in distinguishing between 3 possible return types (`Path`, `File`, and `String`). IMO the Javadoc should state a preference and/or justification for which return values are preferred and why, e.g. "All methods dealing with directory or file paths return a Path, unless otherwise indicated with a `String` or `File` suffix."

For increased consistency, I wonder if converting all File return-type methods to return a Path instead might help reduce the mental burden on users of the class. Asking users to call .toFile() when they want to get a File isn't too much of an inconvenience, especially if it isn't obvious to this class which one is preferred.

Reply:
I usually think of a file handle as the object wrapper (e.g. Java File object) around a (filesystem) file descriptor. But I wasn't attempting to make any particular distinction here, so I'll remove the "handles" to avoid confusion.

Reply:
Good point about the varying return types. I'll give it a pass to make them more consistent on Path.

Reply:
Changed all file/directory related methods in GlobalContext and WorkspaceContext to use Path as return type.